include/ocpp1_6/charge_point.hpp

```diff
@@ -328,7 +328,7 @@ class ChargePoint {
 
     /// EV/EVSE indicates that an error with the given \p error_code occured
     /// \returns true if this state change was possible
-    bool error(int32_t connector, ChargePointErrorCode error_code);
+    bool error(int32_t connector, json error_code);
```

Comment:
Why did the type of error_code change here? I would favor the more specific type of ChargePointErrorCode

Reply:
We receive different error codes from the evse manager. At some time we have to unify or map them

Reply:
so the error_code function is a handler thats called from the evse manager?

Reply:
its called from the OCPP module in the everest-core as a callback when the evse manager sends an error

Reply:
I see, but I would argue, that the one, who is calling this error function should do the conversion of json to ChargePointErrorCode.  The libocpp should not be in charge of knowing about evse manager naming of variables.  So propbably on could do this conversion inside the ocpp module of everest (that would be the right place of conversion).  For example. libocpp should have absolutely no knowlegde about "SLAC"

Reply:
yes definitely, ill keep this PR open untils this + the clearance of the error is fixed

Reply:
ok

Reply:
fixed it - should be merged together with https://github.com/EVerest/everest-core/pull/73